script/hassfest/coverage.py

```diff
@@ -64,18 +74,17 @@ def validate(integrations: dict[str, Integration], config: Config) -> None:
                 not_found.append(line)
                 continue
 
-            if (
-                not line.startswith("homeassistant/components/")
-                or len(path.parts) != 4
-                or path.parts[-1] != "*"
-            ):
+            if not line.startswith("homeassistant/components/") or len(path.parts) != 4:
                 continue
 
             integration_path = path.parent
 
             integration = integrations[integration_path.name]
 
             for check in DONT_IGNORE:
+                if path.parts[-1] not in {"*", check}:
+                    continue
+
```

Comment:
This ensure that either the line ends with wildcard OR the line ends with the current ignore file.